lineapy/cli/cli.py

```diff
@@ -371,6 +373,106 @@ def setup_ipython_dir() -> None:
     os.environ["IPYTHONDIR"] = ipython_dir_name
 
 
+@linea_cli.group("annotate")
+def annotations():
+    """
+    This serves as the command group for all annotate commands.
+    For example:
+        lineapy annotate add . . .
+        lineapy annotate list . . .
+    """
+    pass
+
+
+def validate_annotations_path(ctx, param, value: pathlib.Path):
+    if value.suffix != ".yaml":
+        raise click.BadParameter("path must be a yaml file")
+    return value
+
+
+@annotations.command("add")
+@click.argument(
+    "path",
+    type=click.Path(dir_okay=False, path_type=pathlib.Path),
+    callback=validate_annotations_path,
+)
+@click.option(
+    "--name",
+    "-n",
+    default=None,
+    help="What to name imported resource (do not include .yaml extension). Input file name is used as default",
+    type=str,
+)
+def add(path: pathlib.Path, name: str):
+    """
+    Import user-defined function annotations into Lineapy.
+
+    This command copies the yaml file whose path is provided by the user into the user's .lineapy directory to allow Lineapy to manage it.
+    """
+    # Calculate annotations resource name
+    name = name or path.stem
+    name = name.strip()
+
+    # Create custom annotations folder in user's .lineapy directory to store imported annotations.
+    custom_annotations_dir = linea_folder() / CUSTOM_ANNOTATIONS_FOLDER_NAME
+    if not os.path.exists(custom_annotations_dir):
+        logger.info(
+            f"{custom_annotations_dir} does not exist. Creating it now."
+        )
+        try:
+            os.makedirs(custom_annotations_dir)
+        except OSError as e:
+            logger.error(
+                f"Failed to create {custom_annotations_dir}.\n{str(e)}"
+            )
+            sys.exit(1)
+    else:
+        logger.info(f"{custom_annotations_dir} already exists.")
+
+    # Path to copy destination in user's .lineapy directory
+    destination_file = custom_annotations_dir / (name + ".yaml")
```

Comment:
we need to sanitize this input before using it in file name.

Reply:
also keep the format of xxxx.annotations.yaml since we already use that for existing files. It make it stricter and keep the options of adding other non-annotations yaml files in the future.